### ISSUE TITLE
Argument to ion_rs::value::Builder::new_blob, new_clob functions are &'static

### ISSUE THREAD

**Issue by dlurton** [amazon-ion/ion-rust#465]
https://github.com/amazon-ion/ion-rust/blob/93da7e470e1e194a5e8fb14265a4eea2a69614e8/src/value/mod.rs#L552-L556

Which makes them pretty unusable in my case and I think most other cases as well.   This is a big problem for my application because since we use a lot of blobs.

To demonstrate:

```
use ion_rs::value::Builder;
use ion_rs::value::owned::Element;

fn main() {
    let blob_elem = Element::new_blob("this compiles fine -- arg is already &'static ".as_bytes().into());
    println!("{:?}", blob_elem);

    let hello_world: Box<[u8]> = "hello, world!".as_bytes().into();
    let blob_elem = Element::new_blob(hello_world.as_ref());
    // |                              ^^^^^^^^^^^^^^^^^^^^
    // |                              |
    // |                              borrowed value does not live long enough
    // |                              argument requires that `hello_world` is borrowed for `'static`
    println!("{:?}", blob_elem);
}
```

Note that if you replace the argument with a string literal, i.e. `Element::new_blob("hello, world!".into())`, it works fine.

**Comment by dlurton** [comment]
And while you're fixing this you might want to consider changing the `bytes` argument to be `T: Into<Vec<u8>>` as well.

**Comment by zslayton** [comment]
This was fixed in #479 and will be included in the next release.
